structs/map.go

```diff
@@ -0,0 +1,68 @@
+package structs
+
+import (
+	"errors"
+	"fmt"
+	"github.com/fatih/structs"
+	"reflect"
+)
+
+var (
+	// ErrTagEmpty is returned when an empty tag is passed to ToMap.
+	ErrTagEmpty = errors.New("tag cannot be empty")
+	// ErrInvalidInputType is returned when ToMap receives an input that is not a struct or a pointer to a struct.
+	ErrInvalidInputType = errors.New("invalid input type")
+)
+
+// ToMap converts the given struct to a map[string]any. It uses the "structs" tag to identify the key for each map field.
+// If no tag is present in the Struct field, the field is omitted.
+//
+// The default key string is the struct field name but can be changed in the struct field's tag value.
```

Comment:
Since we've added examples for everything else, care to add an example for the base case? This is mainly to illustrate how the keys is **exactly** the name of the field, and is not changed in any form (e.g. transformed to snake case.